### DISCUSSION TITLE
Several commands in a row

### DISCUSSION THREAD

**Discussion by fion-private** [alandtse/alexa_media_player#2769] [Q&A]
I want to send several voice commands in a row to an Echo from an automation (e.g. flash briefing, weather). To do this, I use the media_player.play_media service. However, the output for the first command is aborted after 1-2 seconds and the system jumps to the next command. This is how the actions were configured:

`alias: How is the weather today
action: media_player.play_media
target:
  entity_id:
    - media_player.echo_fion
data:
  media_content_type: custom
  media_content_id: How is the weather today?
enabled: true`

Is there a way to wait until the first output is finished before starting the next output?

**Comment by danielbrunt57** [comment]
No, it's an Amazon shortcoming.

You'd have to somehow estimate how long it could take and insert a delay before you issue next command. I had a similar issue with multiple TTS commands in my good morning & good night scripts. I rewrote everything to funnel all notify via TTS through a single script with run mode single, max 20.  That script then counts the words, multiplies by 0.57 (trial and error for English) and delays that many seconds before executing the next call and it works great, for my use case. But a flash briefing?  I think that is unsolvable. The undocumented API does not report a completed status message. I highly expect the app would cause the device to abort what it's speaking and start the next, if you were to run multiple routines one right after one another to say something.

**Comment by danielbrunt57** [comment]
A while ago when I was trying to figure something out, I remember wondering if an attribute of the the echo device could assist. That attribute is a timestamp named something like media position updated. I'm on my phone or I'd try and be more specific. I think you can wait for trigger of an attribute change?  I'd watch the device in states and see if the attribute I'm thinking of updates when it finishes speaking.

Let me know what you see...
